code/modules/mob/living/say.dm

```diff
@@ -62,6 +62,24 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 /proc/prefix_to_channel(prefix)
 	return GLOB.department_radio_keys[prefix]
 
+/proc/filter_message(user, message)
+	var/list/banned_words = CONFIG_GET(str_list/word_filter)
+	var/message_lowercase = lowertext(message)
+
+	for(var/word in banned_words)
+		if(findtext(message_lowercase, regex(lowertext(word))))
```

Comment:
For the sake of performance, you should be doing this differently.
The config should be a single regex instead of a list of banned words. As such, the code should be changed to
```
    if(findtext(message_lowercase, regex(config_entry_here)))
```
And the config entry should be a single string stead of a string list. Additionally, don't lowertext() the naughty word regex, that can be handled in the config.

The performance gains are noticeable. At a list length of two words, your code can run ~46560 times in a tick. Mine can run ~96342 times in a tick. The difference in perf becomes even larger as the word list grows, with a list length of seven words being ~19670 and ~79061 times in a tick, respectively.